test/unit/cmap/auth/mongodb_oidc/callback_lock_cache.test.ts

```diff
@@ -0,0 +1,144 @@
+import { expect } from 'chai';
+import * as sinon from 'sinon';
+import { setTimeout } from 'timers';
+
+import {
+  CallbackLockCache,
+  Connection,
+  MongoCredentials,
+  MongoInvalidArgumentError
+} from '../../../../mongodb';
+
+describe('CallbackLockCache', function () {
+  describe('#getCallbacks', function () {
+    const connection = sinon.createStubInstance(Connection);
+    connection.address = 'localhost:27017';
+
+    context('when a request callback does not exist', function () {
+      const credentials = new MongoCredentials({
+        username: 'test_user',
+        password: 'pwd',
+        source: '$external',
+        mechanismProperties: {}
+      });
+      const cache = new CallbackLockCache();
+
+      it('raises an error', function () {
+        try {
+          cache.getCallbacks(connection, credentials);
```

Comment:
This test does not fail if it *does not* raise an error
We've typically done:
```ts
const error = cache.getCallbacks(connection, credentials).catch(error => error);
expect(error).to.be.instanceOf(MongoInvalidArgumentError);
```
With the assumption that our APIs would never _return_ an instance of an error so we'd fail if the call succeeds.

Reply:
`getCallbacks` doesn't return a promise (it's sync) so I added a fail assertion after the call in the try block.